spec/event_tag_utils_spec.rb

```diff
@@ -21,72 +21,88 @@
 require 'optimizely/helpers/event_tag_utils'
 
 describe 'EventTagUtils' do
-  let(:logger) { Optimizely::NoOpLogger.new }
+  let(:logger) { Optimizely::SimpleLogger.new }
 
   describe '.get_revenue_value' do
     it 'should return nil if argument is not a Hash' do
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(nil)).to be_nil
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(0.5)).to be_nil
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(65_536)).to be_nil
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(9_223_372_036_854_775_807)).to be_nil
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value('65536')).to be_nil
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(false)).to be_nil
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(true)).to be_nil
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value([])).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(nil, logger)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(0.5, logger)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(65_536, logger)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(9_223_372_036_854_775_807, logger)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value('65536', logger)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(false, logger)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(true, logger)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value([], logger)).to be_nil
     end
     it 'should return nil if event tags does not contain the revenue' do
       event_tags = {
         'non-revenue' => 5432
       }
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags)).to be_nil
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to be_nil
     end
     it 'should return nil if event tags contains the revenue with a string value' do
       event_tags = {
         'revenue' => 'string'
       }
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags)).to be_nil
+      expect(logger).to receive(:log)
+        .with(Logger::WARN, 'Failed to parse revenue value string from event tags.')
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to be_nil
     end
     it 'should return nil if event tags contains the revenue with a string value' do
       event_tags = {
         'revenue' => '65536'
       }
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags)).to be_nil
+      expect(logger).to receive(:log)
+        .with(Logger::WARN, 'Failed to parse revenue value 65536 from event tags.')
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to be_nil
```

Comment:
No, no no. You failed to understand the main purpose of parsing revenue value. Now we do accept strings and float. Please take a look at PHP PR again. See all the changes carefully.